apps/teams-test-app/src/components/Links.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { ReactElement } from 'react';
+
+import { ApiWithTextInput } from './utils';
+
+const MoveToLink = (): ReactElement =>
+  ApiWithTextInput<string>({
+    name: 'moveToLink',
+    title: 'Move to Link',
+    onClick: {
+      validateInput: input => {
+        if (typeof input !== 'string') {
+          throw new Error('Input should be a string');
+        }
+      },
+      submit: async input => {
+        window.location.href = input;
+        return Promise.resolve('Moved to new link');
```

Comment:
Does this work? I mean I assume it must work since you wrote it, but it's fascinating that redirecting the browser to a new site lets the code keep running so that it resolves the promise?

Reply:
haha, yes it does but its unpredictable. Till the browser does not move to new webpage. We would have short span of time to notice.